pkg/featuregate/featuregate.go

```diff
@@ -40,6 +40,13 @@ var (
 		featuregate.WithRegisterDescription("enables feature to set GOMEMLIMIT and GOMAXPROCS automatically"),
 		featuregate.WithRegisterFromVersion("v0.100.0"),
 	)
+	// EnableTargetAllocatorMTLS is the feature gate that enables mTLS between the target allocator and the collector.
+	EnableTargetAllocatorMTLS = featuregate.GlobalRegistry().MustRegister(
+		"operator.targetallocator.mtls",
+		featuregate.StageAlpha,
+		featuregate.WithRegisterDescription("enables mTLS between the target allocator and the collector"),
+		featuregate.WithRegisterFromVersion("v0.108.0"),
```

Comment:
```suggestion
		featuregate.WithRegisterFromVersion("v0.109.0"),
```

Reply:
Why do we need to enable this only from v0.109.0 if the collector version needed is v0.108.0?

Reply:
This is the operator version which introduces the flag. It has nothing to do with collector versions.